src/lib/translators/mongo.ts

```diff
@@ -1946,13 +1946,52 @@ export class Mongo36Translator extends BaseTranslator {
   }
 
   formula(step: Readonly<S.FormulaStep>): MongoStep {
-    return {
-      $addFields: {
-        [step.new_column]: buildMongoFormulaTree(
-          buildFormulaTree(step.formula, BaseTranslator.variableDelimiters),
-        ),
-      },
+    const mongoFormulaTree: MongoStep | string | number = buildMongoFormulaTree(
+      buildFormulaTree(step.formula, BaseTranslator.variableDelimiters),
+    );
+
+    const denominators: MongoStep[] | string[] | number[] = [{}];
+
+    const iterate = (obj: MongoStep | string | number) => {
+      if (typeof obj === 'object') {
+        Object.keys(obj).forEach(key => {
+          // Add denominator only if it is a Column or a ParenthesisNode
+          if (key == '$divide' && typeof obj[key][1] != 'number') {
+            denominators.push(obj[key][1]);
+          }
+          if (typeof obj[key] === 'object') {
+            iterate(obj[key]);
+          }
+        });
+      }
     };
+
+    // The first element was set for type reason
+    denominators.shift();
+
+    iterate(mongoFormulaTree);
+
+    if (denominators.length == 0) {
+      return {
+        $addFields: {
+          [step.new_column]: mongoFormulaTree,
+        },
+      };
+    } else {
+      const conditions = [];
+
+      for (const d in denominators) {
+        conditions.push({ $in: [denominators[d], [0, null]] });
+      }
```

Comment:
```suggestion
      const conditions = denominators.map((d) => ({ $in: [d, [0, null]] }));
```

Bit weird to use `for...in` for iterating over arrays, right ? :sweat_smile: